src/reporters/composite_reporter.js

```diff
@@ -33,24 +33,50 @@ export default class CompositeReporter {
     }
 
     report(span: Span): void {
-        for (let i = 0; i < this._reporters.length; i++) {
-            let reporter = this._reporters[i];
-            reporter.report(span);
-        }
+        this._reporters.map((r) => {
+            r.report(span);
+        });
```

Comment:
is this decision based on profiling? I'd be surprised if a monad is faster than indexed array access.